python/strings_udf/strings_udf/lowering.py

```diff
@@ -231,6 +231,9 @@ def id_func_impl(context, builder, sig, args):
             # Lookup table required for conversion functions
             # must be resolved at runtime after context initialization,
             # therefore cannot be a global variable
+            global character_flags_table_ptr
+            if character_flags_table_ptr is None:
+                character_flags_table_ptr = get_character_flags_table_ptr()
```

Comment:
Can you explain why this is required? The comment right above says `cannot be a global variable`